pkg/aggregate/iterator.go

```diff
@@ -71,7 +71,7 @@ func (as *AggregateSeries) CanAggregate(partitionAggr AggrType) bool {
 	// make sure the DB has all the aggregators we need (on bits in the mask)
 	// and that the requested interval is greater/eq to aggregator resolution and is an even divisor
 	return ((aggrMask & partitionAggr) == aggrMask) &&
-		as.interval >= as.rollupTime && (as.interval%as.rollupTime == 0)
+		as.interval >= as.rollupTime && (as.interval%as.rollupTime == 0 || as.interval/as.rollupTime > 3)
```

Comment:
What exactly are you checking here?

Reply:
@imakhlin explained to Tal, its handling prom step rounding errors issues

Reply:
Better add a comment that explains that condition, this condition isn't really self explanatory